BetterPasswordPlugin.inc.php

```diff
@@ -160,24 +160,52 @@ function checkPassword($hookName, $args) {
 		if ($this->getSetting(CONTEXT_SITE, 'betterPasswordCheckBlacklist')) {
 			$badPassword = false;
 			$lowerPassword = strtolower($password);
-			foreach ($this->getBlacklists() as $filename) {
-				/* In case of out-of-memory error, break glass
-				$passwordfile = fopen($filename, "r");
-				while (!feof($passwordfile)) {
-					$disallowed = fgets($passwordfile);
-					if ($lowerPassword === $disallowed) {
-						$badPassword = true;
-						break;
-					}
-				}
-				fclose($passwordfile);
-				 */
-				$passwordfile = file_get_contents($filename);
-				if (strpos($passwordfile, $lowerPassword) !== false) {
+                        $sha_pass = sha1($lowerPassword);
+                        $password_hash = substr(sha1($lowerPassword),0,2);
+                        $blacklistFileSetting = $this->getSetting(CONTEXT_SITE, 'betterPasswordBlacklistFiles');
+                        if (is_null($blacklistFileSetting)) {
+                            $blacklistFileSettingHash = array();
+                            foreach ($this->getBlacklists() as $filename) {
+                                $blacklistFileSettingHash[$filename] = sha1_file($filename);
+                            }
+                            $this->updateSetting(CONTEXT_SITE, 'betterPasswordBlacklistFiles', $blacklistFileSettingHash);
+                        }
+                        else {
+                            $currBlacklist = array();
+                            foreach ($this->getBlacklists() as $filename) {
+                                $currBlacklist[$filename] = sha1_file($filename);
+                            }
+                            $prevBlacklist = $this->getSetting(CONTEXT_SITE, 'betterPasswordBlacklistFiles');
+                            $t = empty(array_udiff_uassoc($currBlacklist, $prevBlacklist, "strcmp", "strcmp"));
+                            $s = empty(array_udiff_uassoc($prevBlacklist, $currBlacklist, "strcmp", "strcmp"));
+                            
+                            if (!empty(array_udiff_uassoc($currBlacklist, $prevBlacklist, "strcmp", "strcmp")) || !empty(array_udiff_uassoc($prevBlacklist, $currBlacklist, "strcmp", "strcmp"))) {
+                                $tempFilename = $this->handleTempFile();
+                                $fpTemp = fopen($tempFilename, 'w');
+                                foreach ($this->getBlacklists() as $filename) {
```

Comment:
I'm not sure what we are aiming for here.  If `$tempFileName` already contains all passwords which are longer than the site minimum, why not just read the passwords directly from that file?  Is this leftover code until 198?